test-suite/Math/NumberTheory/ArithmeticFunctionsTests.hs

```diff
@@ -270,6 +272,45 @@ mangoldtOeis = oeisAssertion "A014963" expMangoldtA
   , 73, 1, 1, 1, 1, 1, 79, 1, 3, 1, 83, 1, 1, 1, 1, 1, 89, 1, 1, 1, 1, 1, 1
   ]
 
+nFreedomProperty1 :: Word -> NonZero Natural -> Bool
+nFreedomProperty1 n (NonZero m) =
+    isNFree n m == (all ((< n) . snd) . factorise) m
+
+nFreedomProperty2 :: NonZero Word -> NonNegative Int -> Bool
+nFreedomProperty2 (NonZero n) (NonNegative m) =
+    let n' | n == maxBound = n
+           | otherwise     = n + 1
+    in take m (filter (isNFree n') [1 ..]) == take m (nFrees n' :: [Integer])
+
+nFreedomProperty3 :: NonZero Word -> Positive Int -> Bool
+nFreedomProperty3 (NonZero n) (Positive m) =
+    let n' | n == maxBound = n
+           | otherwise     = n + 1
+        zet = 1 / zetas 1e-14 !! (fromIntegral n') :: Double
+        m' = 100 * m
+        nfree = fromIntegral m' / fromIntegral (last (take m' $ nFrees n' :: [Integer]))
+    in 1 / fromIntegral m >= abs (zet - nfree)
+
+-- |
+-- * Using a bounded integer type like @Int@ instead of @Integer@ here means
+-- even a relatively low value of @n@, e.g. 20 may cause out-of-bounds memory
+-- accesses in @nFreesBlock@.
+-- * Using @Integer@ prevents this, so that is the numeric type used here.
+nFreesBlockProperty1 :: NonZero Word -> Positive Integer -> Word -> Bool
```

Comment:
There is Math.NumberTheory.TestUtils.Wrappers.Power, which seems more suitable than NonZero here.